src/hotspot/os/linux/os_linux.cpp

```diff
@@ -259,52 +259,52 @@ julong os::Linux::available_memory() {
     avail_mem = free_memory();
   }
   log_trace(os)("available memory: " JULONG_FORMAT, avail_mem);
-  return avail_mem;
+  return static_cast<size_t>(avail_mem);
 }
 
-julong os::free_memory() {
+size_t os::free_memory() {
   return Linux::free_memory();
 }
 
-julong os::Linux::free_memory() {
+size_t os::Linux::free_memory() {
   // values in struct sysinfo are "unsigned long"
   struct sysinfo si;
   julong free_mem = available_memory_in_container();
   if (free_mem != static_cast<julong>(-1L)) {
     log_trace(os)("free container memory: " JULONG_FORMAT, free_mem);
-    return free_mem;
+    return static_cast<size_t>(free_mem);
   }
 
   sysinfo(&si);
   free_mem = (julong)si.freeram * si.mem_unit;
   log_trace(os)("free memory: " JULONG_FORMAT, free_mem);
-  return free_mem;
+  return static_cast<size_t>(free_mem);
 }
 
-jlong os::total_swap_space() {
+size_t os::total_swap_space() {
   if (OSContainer::is_containerized()) {
     if (OSContainer::memory_limit_in_bytes() > 0) {
-      return (jlong)(OSContainer::memory_and_swap_limit_in_bytes() - OSContainer::memory_limit_in_bytes());
+      return static_cast<size_t>(OSContainer::memory_and_swap_limit_in_bytes() - OSContainer::memory_limit_in_bytes());
     }
   }
   struct sysinfo si;
   int ret = sysinfo(&si);
   if (ret != 0) {
     return -1;
   }
-  return  (jlong)(si.totalswap * si.mem_unit);
+  return  static_cast<size_t>(si.totalswap * si.mem_unit);
 }
 
-static jlong host_free_swap() {
+static size_t host_free_swap() {
```

Comment:
It is not obvious to me that callers of this are reacting properly when the return value goes from signed to unsigned. I'm thinking about code in os::free_swap_space below:
```
jlong host_free_swap_val = MIN2(os::total_swap_space(), host_free_swap());
```
How does MIN2 react to -1 vs 0xFFFFFFFFFFFFFFFF